packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -314,16 +314,33 @@ function useFocusContainment(scopeRef: RefObject<Element[]>, contain: boolean) {
 
       let focusedElement = document.activeElement;
       let scope = scopeRef.current;
-      if (!isElementInScope(focusedElement, scope)) {
+      let childScopeRef = isElementInChildScope(focusedElement, scopeRef);
+      if (!childScopeRef || typeof childScopeRef !== 'object') {
         return;
       }
-
-      let walker = getFocusableTreeWalker(getScopeRoot(scope), {tabbable: true}, scope);
-      walker.currentNode = focusedElement;
-      let nextElement = (e.shiftKey ? walker.previousNode() : walker.nextNode()) as FocusableElement;
-      if (!nextElement) {
-        walker.currentNode = e.shiftKey ? scope[scope.length - 1].nextElementSibling : scope[0].previousElementSibling;
-        nextElement = (e.shiftKey ? walker.previousNode() : walker.nextNode())  as FocusableElement;
+      let focusNextInChildScope = (childScopeRef: ScopeRef): FocusableElement | null => {
+        if (childScopeRef !== scopeRef) {
+          let walker = getFocusableTreeWalker(getScopeRoot(childScopeRef.current), {tabbable: true}, scope, scopeRef);
+          walker.currentNode = focusedElement;
+          let nextElement = (e.shiftKey ? walker.previousNode() : walker.nextNode()) as FocusableElement;
+          if (nextElement) {
+            return nextElement;
+          } else {
+            return focusNextInChildScope(focusScopeTree.getTreeNode(childScopeRef).parent.scopeRef);
+          }
+        }
+        return null;
+      };
+      let nextElement = focusNextInChildScope(childScopeRef);
+
+      if (!nextElement && isElementInScope(focusedElement, scope)) {
+        let walker = getFocusableTreeWalker(getScopeRoot(scope), {tabbable: true}, scope);
+        walker.currentNode = focusedElement;
+        nextElement = (e.shiftKey ? walker.previousNode() : walker.nextNode()) as FocusableElement;
+        if (!nextElement) {
+          walker.currentNode = e.shiftKey ? scope[scope.length - 1].nextElementSibling : scope[0].previousElementSibling;
+          nextElement = (e.shiftKey ? walker.previousNode() : walker.nextNode()) as FocusableElement;
+        }
```

Comment:
noted that shift tabbing from the sub dialog link actually closed the entire menu instead of moving focus back to the original menu item. Not sure if this section of the code is the reason or if it is the tracked restoreFocus node

Reply:
just saw that there is a test that explicitly checks that the whole menu closes on shift tabbing. Why is there a difference between shift-tabbing (closes the entire menu) and tabbing forward (refocuses the original menu item)?

Reply:
Yeah, so this section of code is never run when you shift tab out of a sub dialog
That's all in restoreFocus

What happens is, Shift+Tab
we get the nextElement, which is the previousNode, which is the li we came from, the same node as nodeToRestore
that element isn't in scope, so we set the walker to the nodeToRestore
then, we back the walker up once more, which takes us out to the menu trigger

the reason for this, is so that shift + tab out of a menu takes you to the item before the trigger, see
https://reactspectrum.blob.core.windows.net/reactspectrum/b810c96d7b6fa4e404d5309f475cddf19d166f0e/storybook/index.html?path=/story/menutrigger--menu-closes-on-blur&providerSwitcher-express=false

however, while writing it down, I just realized that in this particular case, I could inadvertently go to a later element than the trigger, so I'll push a change to it